packages/camera/camera_windows/windows/camera_plugin.h

```diff
@@ -93,6 +95,7 @@ class CameraPlugin : public flutter::Plugin,
   flutter::TextureRegistrar* texture_registrar_;
   flutter::BinaryMessenger* messenger_;
   std::vector<std::unique_ptr<Camera>> cameras_;
+  std::shared_ptr<TaskRunner> task_runner_;
```

Comment:
Why is this a `shared_ptr` rather than a `unique_ptr`. This class controls `Camera` lifetimes; in what scenario would the task runner need to outlive it?

Reply:
The instance is shared with the [cameras](https://github.com/CodemateLtd/packages/blob/272013224735c2cc9411227907df83b89ec6bd56/packages/camera/camera_windows/windows/camera_plugin.cpp#L256-L257) and ultimately with the [`CaptureController`](https://github.com/CodemateLtd/packages/blob/272013224735c2cc9411227907df83b89ec6bd56/packages/camera/camera_windows/windows/camera.cpp#L60-L61).

In terms of lifetime, I _guess_ there might be a small window where IMF is still running the [callback](https://github.com/CodemateLtd/packages/blob/272013224735c2cc9411227907df83b89ec6bd56/packages/camera/camera_windows/windows/capture_controller.cpp#L873-L877) when the plugin is being disposed, though I’m not really sure if that’s actually possible.